game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1127,6 +1379,8 @@ private void fireNavalBombardment(final IDelegateBridge bridge) {
                 attacker, battleSite, gameData));
     // bombarding units can't move after bombarding
     if (!headless) {
+      // TODO: StepRefactor: Why is a change always added even if there are no units?
```

Comment:
We've had some disucssions on TODO. I can provide links if desired. Our net conclusion is that TODOs wind up being confusing and are often never done. It's perhaps better to keep a personal backlog for such questions rather than inject them into the code where it's still only really useful for the author anyways (in which case just having a personal backlog is just-as-well).

Reply:
No follow-up required.

Reply:
I had found the entry in the wiki at https://github.com/triplea-game/triplea/wiki/Java-Code-Conventions#todo-comments-attach-a-tracking-token-5249.  I used "StepRefactor" is my 'grep token'.